test/helpers/envtest.go

```diff
@@ -76,31 +74,42 @@ func newTestEnvironment() *TestEnvironment {
 	}
 }
 
-func NewTestEnvironment() (*TestEnvironment, error) {
-	env := newTestEnvironment()
+type TestEnvironment struct {
+	*envtest.Environment
+	manager.Manager
+	client.Client
+
+	Config  *rest.Config
+	scheme  *runtime.Scheme
+	doneMgr chan struct{}
+}
 
-	cfg, err := env.Environment.Start()
-	if err != nil {
-		return env, err
-	}
+func NewTestEnvironment() *TestEnvironment {
```

Comment:
Does this need to keep track of the number of times this is called, otherwise it looks like any call to `Stop()` may tear down the environment for everyone.

Reply:
Either that or the first envtest environment that is stood up could possibly be treated separately from others, and addtional envtest environments are created as this is called usiing the `UseExistingCluster` flag, which would prevent teardown of the controlplane when `Stop()` is called.

Reply:
`UseExistingCluster` flag uses the local kubeconfig to call a cluster, that's not suitable in this case I think